src/Microsoft.AspNetCore.Localization/ApplicationBuilderExtensions.cs

```diff
@@ -43,12 +43,77 @@ public static IApplicationBuilder UseRequestLocalization(this IApplicationBuilde
             {
                 throw new ArgumentNullException(nameof(app));
             }
+            
             if (options == null)
             {
                 throw new ArgumentNullException(nameof(options));
             }
 
             return app.UseMiddleware<RequestLocalizationMiddleware>(Options.Create(options));
         }
+
+        /// <summary>
+        /// Adds the <see cref="RequestLocalizationMiddleware"/> to automatically set culture information for
+        /// requests based on information provided by the client. 
+        /// </summary>
+        /// <param name="app">The <see cref="IApplicationBuilder"/>.</param>
+        /// <param name="optionsAction"></param>
+        /// <returns>The <see cref="IApplicationBuilder"/>.</returns>
+        public static IApplicationBuilder UseRequestLocalization(
+            this IApplicationBuilder app,
+            Action<RequestLocalizationOptions> optionsAction)
+        {
+            if (app == null)
+            {
+                throw new ArgumentNullException(nameof(app));
+            }
+
+            if (optionsAction == null)
+            {
+                throw new ArgumentNullException(nameof(optionsAction));
+            }
+
+            var options = new RequestLocalizationOptions();
+            optionsAction.Invoke(options);
+
+            return app.UseMiddleware<RequestLocalizationMiddleware>(Options.Create(options));
+        }
+
+        /// <summary>
+        /// Adds the <see cref="RequestLocalizationMiddleware"/> to automatically set culture information for
+        /// requests based on information provided by the client. 
+        /// </summary>
+        /// <param name="app">The <see cref="IApplicationBuilder"/>.</param>
+        /// <param name="cultures">The culture names to be added by the application, which is represents both supported cultures and UI cultures.</param>
+        /// <returns>The <see cref="IApplicationBuilder"/>.</returns>
+        /// <remarks>
+        /// Note that the first culture is the default culture name.
+        /// </remarks>
+        public static IApplicationBuilder UseRequestLocalization(
+            this IApplicationBuilder app,
+            params string[] cultures)
+        {
+            if (app == null)
+            {
+                throw new ArgumentNullException(nameof(app));
+            }
+
+            if (cultures == null)
+            {
+                throw new ArgumentNullException(nameof(cultures));
+            }
+
+            if (cultures.Length == 0)
+            {
+                throw new ArgumentException($"The {cultures} cannot be null.", nameof(cultures));
```

Comment:
I 'll move this to the `Resources.resx` at the end

Reply:
This isn't the right text. It should say something like "Please provide at least one culture."